Frontend/ui-customer/src/Components/SettingSideBar/View.purs

```diff
@@ -207,9 +207,12 @@ profileView state push =
         , orientation HORIZONTAL
         , gravity CENTER_VERTICAL
         , margin $ MarginTop 4
-        , visibility case profileCompleteValue state of
-            "100" -> GONE
-            _ -> VISIBLE
+        , visibility $ 
+              if state.appConfig.showProfileStatus 
+                then case profileCompleteValue state of
+                    "100" -> GONE
+                    _ -> VISIBLE
+                else GONE
```

Comment:
keep logic in a function separately and it can be just like - 

```purescript
       case state.appConfig.showProfileStatus, profileCompleteValue state of
            true , "100" -> GONE
            false, _     -> GONE
            _, _         -> VISIBLE
```